src/arrangement.cpp

```diff
@@ -30,11 +30,22 @@ typedef CGAL::Cartesian_converter<Kernel,Rat_kernel>    Kernel_to_Rational;
 
 typedef PySequenceCirculator<Ccb_halfedge_circulator> PyCcbHalfedgeCirculator;
 
+
 void insert_segments_in_arr(Segment_Arrangement_2& arr, std::vector<Segment_2>& segs) {
+	try{
     CGAL::insert(arr, segs.begin(), segs.end());
+	} catch (CGAL::Precondition_exception)
```

Comment:
It migth be better to translate this exception to python with pybind11, no?

Reply:
Hi, yes apologies, I submitted in a rush and there are parts of the code (including the skeleton part below) that remain from previous tests I did + merging with the master branch. Happy to create a new pull request with the relevant changes only. 
Also, Indeed would be much better to translate the exception with pybind. This is a workaround due to my inexperience with pybind (more familiar with swig)